### PR TITLE
[enhancement](jni)append exception check for jni code.

### PR DESCRIPTION
**Pull request by hubgeter**
## Proposed changes
Because in jni, we need to manually check for exceptions thrown, and if we do not handle the exception, the program will continue to execute, which is not like the try catch logic we often use. Therefore, in order to reduce the potential risks caused by not handling exceptions, I checked most of the jni code and added exception detection for it.

### CODE REVIEW

**Comment by hubgeter** [comment]
run buildall

File: be/src/util/jvm_metrics.cpp [hunk 1]
```diff
@@ -90,9 +92,12 @@ JvmMetrics::JvmMetrics(MetricRegistry* registry, JNIEnv* env) {
             break;
         }
         try {
-            _jvm_stats.init(env);
+            Status st = _jvm_stats.init();
+            if (!st) {
+                LOG(WARNING) << "jvm Stats Init Fail. " << st.to_string();
```
Comment (morningman) [inline be/src/util/jvm_metrics.cpp:97]: Should we call `break` if `!st`?

File: be/src/util/jvm_metrics.cpp [hunk 2]
```diff
@@ -182,193 +188,260 @@ void JvmMetrics::update() {
...
-    env = ENV;
-    _managementFactoryClass = env->FindClass("java/lang/management/ManagementFactory");
-    if (_managementFactoryClass == nullptr) {
-        LOG(WARNING)
-                << "Class java/lang/management/ManagementFactory Not Find.JVM monitoring fails.";
-        return;
-    }
+Status JvmStats::init() {
+    JNIEnv* env = nullptr;
+    RETURN_IF_ERROR(JniUtil::GetJNIEnv(&env));
```
Comment (morningman) [inline be/src/util/jvm_metrics.cpp:193]: Why do we need to get `env` again here?

**Comment by hubgeter** [comment]
run buildall

File: be/src/util/jvm_metrics.cpp (hunk 2 above)
Comment (hubgeter) [reply be/src/util/jvm_metrics.cpp]: There is really no need

File: be/src/util/jvm_metrics.cpp (hunk 1 above)
Comment (hubgeter) [reply be/src/util/jvm_metrics.cpp:97]: It doesn't matter if it is needed or not, because I will check `_jvm_stats.init_complete()`.

**Review by morningman — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**